### PR TITLE
[Enhance] Support mim

### PR DESCRIPTION
**Pull request by wHao-Wu**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation
* Support MIM

### CODE REVIEW

File: MANIFEST.in [hunk 1]
```diff
@@ -1,6 +1,6 @@
...
+include mmdet3d/.mim/model-index.yml
 include requirements/*.txt
-recursive-include mmdet3d/configs *.py *.yml
-recursive-include mmdet3d/tools *.sh *.py
-recursive-include mmdet3d/ops *.cpp *.cu *.h *.cc
-recursive-include mmdet3d/demo *.png *.jpg *.bin *.pkl *.py
+recursive-include mmdet3d/.mim/ops *.cpp *.cu *.h *.cc
+recursive-include mmdet3d/.mim/configs *.py *.yml
+recursive-include mmdet3d/.mim/tools *.sh *.py
+recursive-include mmdet3d/.mim/demo *.png *.jpg *.bin *.pkl *.py
```
Comment (ZwwWayne) [inline MANIFEST.in:6]: Do we really need to include demo? Because it will increase the size of the package.

**Comment by ZwwWayne** [comment]
May test the compatibility with MIM.

File: MANIFEST.in (hunk 1 above)
Comment (wHao-Wu) [reply MANIFEST.in]: Sure about that. Besides, `demo` is rarely used when developing or researching. I have removed it from `MANIFEST.in`.

File: setup.py [hunk 2]
```diff
@@ -135,7 +139,56 @@ def gen_packages_items():
...
+        # installed by `pip install -e .`
+        mode = 'symlink'
+    elif 'sdist' in sys.argv or 'bdist_wheel' in sys.argv:
+        # installed by `pip install .`
+        # or create source distribution by `python setup.py sdist`
+        mode = 'copy'
+    else:
+        return
+
+    filenames = ['tools', 'configs', 'demo', 'model-index.yml']
```
Comment (zhouzaida) [inline setup.py:161]:
```suggestion
    filenames = ['tools', 'configs', 'model-index.yml']
```

Comment (wHao-Wu) [reply setup.py]: Thx. I have updated the `setup.py`.

**Comment by wHao-Wu** [comment]
@ZwwWayne @zhouzaida 
The test of compatibility between mmdet3d and mim is completed successfully.

**Review by zhouzaida — APPROVED** [review]
LTGM

**Pull request merged**